tools/json2bin/event.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+import json
+import pathlib
+import sys
+
+from convert import pad, u16, u32
+
+def parse_bg_event(bg: dict) -> bytes:
+    return b''.join([
+        u16(bg['script']),
+        u16(bg['type']),
+        u32(bg['x']),
+        u32(bg['z']),
+        u32(bg['y']),
+        u16(bg['player_facing_dir']),
+        pad(2),
+    ])
+
+def parse_bg_events(bg_events: list[dict]) -> bytes:
+    return b''.join([
+        u32(len(bg_events)),
+        *[parse_bg_event(bg) for bg in bg_events],
+    ])
+
+def parse_object_event(obj: dict, i: int) -> bytes:
+    obj_data = obj['data'] + [0] * (3 - len(obj['data'])) # pad up to length 3
```

Comment:
Just making a note that I *did* test this in the Python REPL by trying to pad with a negative result size, and Python gracefully refused to alter it:

```
>>> l = []
>>> l += [0] * 3
>>> l
[0, 0, 0]
>>> l += [0] * -5
>>> l
[0, 0, 0]
```